src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -28,164 +23,134 @@ import * as ReceiptUtils from '@libs/ReceiptUtils';
+    transactions: OnyxCollection<Transaction>;
 
-    /** Active IOU Report for current report */
-    iouReport: PropTypes.shape({
-        /** AccountID of the manager in this iou report */
-        managerID: PropTypes.number,
+    /** All of the transaction violations */
+    transactionViolations: OnyxCollection<TransactionViolations>;
+};
 
-        /** AccountID of the creator of this iou report */
-        ownerAccountID: PropTypes.number,
+type ReportPreviewProps = ReportPreviewOnyxProps & {
+    /** All the data of the action */
+    action: ReportAction;
 
-        /** Outstanding amount in cents of this transaction */
-        total: PropTypes.number,
+    /** The associated chatReport */
+    chatReportID: string;
 
-        /** Currency of outstanding amount of this transaction */
-        currency: PropTypes.string,
+    /** The active IOUReport, used for Onyx subscription */
+    iouReportID: string;
 
-        /** Is the iouReport waiting for the submitter to add a credit bank account? */
-        isWaitingOnBankAccount: PropTypes.bool,
-    }),
+    /** The report's policyID, used for Onyx subscription */
+    policyID: string;
 
-    /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user accountID */
-        accountID: PropTypes.number,
-    }),
+    /** Extra styles to pass to View wrapper */
+    containerStyles?: StyleProp<ViewStyle>;
 
     /** Popover context menu anchor, used for showing context menu */
-    contextMenuAnchor: refPropTypes,
+    contextMenuAnchor?: ContextMenuAnchor;
 
     /** Callback for updating context menu active state, used for showing context menu */
-    checkIfContextMenuActive: PropTypes.func,
+    checkIfContextMenuActive?: () => void;
 
     /** Whether a message is a whisper */
-    isWhisper: PropTypes.bool,
-
-    /** All the transactions, used to update ReportPreview label and status */
-    transactions: PropTypes.objectOf(transactionPropTypes),
+    isWhisper?: boolean;
 
-    /** All of the transaction violations */
-    transactionViolations: transactionViolationsPropType,
-
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    contextMenuAnchor: null,
-    chatReport: {},
-    containerStyles: [],
-    iouReport: {},
-    checkIfContextMenuActive: () => {},
-    session: {
-        accountID: null,
-    },
-    isWhisper: false,
-    transactionViolations: {
-        violations: [],
-    },
-    policy: {
-        isHarvestingEnabled: false,
-    },
-    transactions: {},
+    /** Whether the corresponding report action item is hovered */
+    isHovered?: boolean;
 };
 
-function ReportPreview(props) {
+function ReportPreview({
+    iouReport,
+    session,
+    policy,
+    iouReportID,
+    policyID,
+    chatReportID,
+    chatReport,
+    action,
+    containerStyles,
+    contextMenuAnchor,
+    transactions,
+    transactionViolations,
```

Comment:
Why don't we keep original default values?
i.e. `iouReport = {}, chatReport = {}`

Reply:
@situchan Since it all are onyx values they have `null` in their typing. That's mean that cases where the value is null will be handled by the app anyway (with optional chaining, conditions), so no need in default values.